include/realtime_tools/realtime_publisher.hpp

```diff
@@ -153,6 +153,9 @@ class RealtimePublisher
 #endif
   }
 
+protected:
+  std::thread thread_;
+
```

Comment:
```suggestion
  std::thread & get_thread() { return thread_; }
  
  const std::thread & get_thread() const { return thread_; }

```

Reply:
This way you don't have to make it protected and without inheritance you might be able to rename it

Reply:
Ah yes, it would work for me :)

Reply:
This was actually my suggestion in #176. 
I think it is not a good idea to expose the thread publicly as it would allow to interact with the class internals
E.g. joining, detaching

Reply:
I understand your point, but inheriting to just have this renaming and other stuff would be cumbersome right?

Moreover, if the user does it with that method, then he is doing something wrong which is not supposed to be. May be we can document it to avoid such things. 

What do you think? @firesurfer?